pkg/oauth/handlers/authenticator.go

```diff
@@ -101,24 +102,30 @@ func (h *accessAuthenticator) HandleAccess(ar *osin.AccessRequest, w http.Respon
 		klog.Warningf("Received unknown access token type: %s", ar.Type)
 	}
 
+	audit.AddUsernameAnnotation(ar.HttpRequest, ar.Username)
```

Comment:
aren't the annotations handled in the above authentication handling code already?

Reply:
Not really. For `authenticator.Request`, we are handling the username inside as this functions contain the logic dismantle the request and generate a username out of it. For the case of `authenticator.Password` the username and password are already known and handed down. So there is no need to have that logic within the `AuthenticatePassword` methods. Especially as there are fewer usages then implementations.

Reply:
8 implementations (+1 in vendor, which I can't influence without changing library-go) vs 3 usages.

Reply:
And `AuthenticationAssertion` and `AuthenticateClient` don't handle username auditing afaik. But they seem to be very edge-case-y.